unison-cli/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -2302,21 +2306,27 @@ showDiffNamespace sn ppe oldPath newPath OBD.BranchDiffOutput {..} =
         (P.syntaxToColor . DeclPrinter.prettyDeclOrBuiltinHeader (HQ'.toHQ hq))
     phq' :: _ -> Pretty = P.syntaxToColor . prettyHashQualified'
     phq :: _ -> Pretty = P.syntaxToColor . prettyHashQualified
-    --
+
     -- DeclPrinter.prettyDeclHeader : HQ -> Either
-    numPatch :: Path.Absolute -> Name -> Numbered Pretty
+    numPatch :: Input.AbsBranchId -> Name -> Numbered Pretty
     numPatch prefix name =
-      addNumberedArg . Name.toString . Name.makeAbsolute $ Path.prefixName prefix name
-
-    numHQ :: Path.Absolute -> HQ.HashQualified Name -> Referent -> Numbered Pretty
-    numHQ prefix hq r = addNumberedArg (HQ.toString hq')
-      where
-        hq' = HQ.requalify (fmap (Name.makeAbsolute . Path.prefixName prefix) hq) r
-
-    numHQ' :: Path.Absolute -> HQ'.HashQualified Name -> Referent -> Numbered Pretty
-    numHQ' prefix hq r = addNumberedArg (HQ'.toString hq')
-      where
-        hq' = HQ'.requalify (fmap (Name.makeAbsolute . Path.prefixName prefix) hq) r
+      addNumberedArg $ prefixBranchId prefix name
+
+    numHQ :: Input.AbsBranchId -> HQ.HashQualified Name -> Referent -> Numbered Pretty
+    numHQ prefix hq r =
+      addNumberedArg . HQ.toStringWith (prefixBranchId prefix) . HQ.requalify hq $ r
+
+    numHQ' :: Input.AbsBranchId -> HQ'.HashQualified Name -> Referent -> Numbered Pretty
+    numHQ' prefix hq r =
+      addNumberedArg . HQ'.toStringWith (prefixBranchId prefix) . HQ'.requalify hq $ r
+
+    -- E.g.
+    -- prefixBranchId "#abcdef" "base.List.map" -> "#abcdef.base.List.map"
+    -- prefixBranchId ".base" "List.map" -> ".base.List.map"
+    prefixBranchId :: Input.AbsBranchId -> Name -> String
+    prefixBranchId branchId name = case branchId of
+      Left sbh -> "#" <> SBH.toString sbh <> Name.toString (Name.makeAbsolute  name)
```

Comment:
```suggestion
      Left sbh -> "#" <> SBH.toString sbh <> ":" <> Name.toString (Name.makeAbsolute  name)
```
Let's do this to match the `pull` syntax for now, even though neither syntax is totally valid in this context, and then we'll figure out a way forward towards whatever goal later.  In any case we should make them match.  We can potentially change the `pull` syntax in the future if we're careful about it.

The `Name` here, I would guess, is always "relative", which makes this transformation feel a little funny to me, but we can deal with that later.

Reply:
It's always relative _to the prefix_, so this transformation should always be valid if the prefix is a SBH 👍🏼